From: Sebastien Delafond <seb@debian.org>
Date: Thu, 29 Jul 2010 19:46:38 +0200
Subject: [PATCH] jruby_home is at a specific location on Debian

---
 bin/jruby.sh |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/bin/jruby.sh b/bin/jruby.sh
index 395da7c..67fe065 100755
--- a/bin/jruby.sh
+++ b/bin/jruby.sh
@@ -47,6 +47,11 @@ else
   JRUBY_HOME=`dirname "$JRUBY_HOME_1"`  # the . dir
 fi
 
+# set JRUBY_HOME to /u/l/jruby except for its own build process
+if [ -z "$BUILDING_JRUBY" ]; then
+  JRUBY_HOME=/usr/lib/jruby/
+fi
+
 if [ -z "$JRUBY_OPTS" ] ; then
   JRUBY_OPTS=""
 fi
-- 
