From: Miguel Landaeta <nomadium@debian.org>
Date: Sun, 24 May 2015 23:54:04 -0300
Subject: Fix some warnings/errors with jruby shell script
Forwarded: no

diff --git a/bin/jruby.sh b/bin/jruby.sh
index f182b75..686f185 100755
--- a/bin/jruby.sh
+++ b/bin/jruby.sh
@@ -116,10 +116,10 @@ if [ "$JRUBY_PARENT_CLASSPATH" != "" ]; then
 else
     # add other jars in lib to CP for command-line execution
     for j in "$JRUBY_HOME"/lib/*.jar; do
-        if [ "$j" == "$JRUBY_HOME"/lib/jruby.jar ]; then
+        if [ "$j" = "$JRUBY_HOME"/lib/jruby.jar ]; then
           continue
         fi
-        if [ "$j" == "$JRUBY_HOME"/lib/jruby-complete.jar ]; then
+        if [ "$j" = "$JRUBY_HOME"/lib/jruby-complete.jar ]; then
           continue
         fi
         if [ "$CP" ]; then
@@ -246,7 +246,7 @@ do
 done
 
 # Force file.encoding to UTF-8 when on Mac, since Apple JDK defaults to MacRoman (JRUBY-3576)
-if [[ -z "$JAVA_ENCODING" ]]; then
+if [ -z "$JAVA_ENCODING" ]; then
   java_args="${java_args} -Dfile.encoding=UTF-8"
 fi
 
