From: Miguel Landaeta <nomadium@debian.org>
Date: Sat, 30 May 2015 12:57:27 -0300
Subject: Disable definitions depending on jruby-openssl since is not packaged yet
Forwarded: no

diff --git a/spec/ruby/core/marshal/fixtures/marshal_data.rb b/spec/ruby/core/marshal/fixtures/marshal_data.rb
index c665e1d..3533d80 100644
--- a/spec/ruby/core/marshal/fixtures/marshal_data.rb
+++ b/spec/ruby/core/marshal/fixtures/marshal_data.rb
@@ -126,21 +126,21 @@ end
 class UserString < String
 end
 
-require 'openssl'
-
-class UserData < OpenSSL::X509::Name
-  alias _dump_data to_a
-
-  def _load_data entries
-    entries.each do |entry|
-      add_entry(*entry)
-    end
-  end
-end
-
-class UserDataUnloadable < UserData
-  undef _load_data
-end
+#require 'openssl'
+#
+#class UserData < OpenSSL::X509::Name
+#  alias _dump_data to_a
+#
+#  def _load_data entries
+#    entries.each do |entry|
+#      add_entry(*entry)
+#    end
+#  end
+#end
+
+#class UserDataUnloadable < UserData
+#  undef _load_data
+#end
 
 module Meths
   def meths_method() end
