From: Miguel Landaeta <nomadium@debian.org>
Date: Sat, 12 Nov 2016 19:41:24 +0000
Subject: Adjust rake to run jruby unit tests in Debian
Forwarded: no

diff --git a/rakelib/rspec.rake b/rakelib/rspec.rake
index dfdf655..fddd1ca 100644
--- a/rakelib/rspec.rake
+++ b/rakelib/rspec.rake
@@ -15,8 +15,9 @@
 
 namespace :spec do
   # Put Rake on the load path for JI specs without requiring rubygems
-  rake_location = File.join(Gem.loaded_specs['rake'].full_gem_path, "lib")
-  gem 'rspec'
+  #rake_location = File.join(Gem.loaded_specs['rake'].full_gem_path, "lib")
+  #gem 'rspec'
+  rake_location = './lib/ruby/1.9/rake.rb'
   require 'rspec/core/rake_task'
 
   compile_flags = {
diff --git a/rakelib/test.rake b/rakelib/test.rake
index 2501336..302e2cc 100644
--- a/rakelib/test.rake
+++ b/rakelib/test.rake
@@ -114,6 +114,7 @@ namespace :test do
     t.ruby_opts << '-I test/externals/ruby1.9'
     t.ruby_opts << '-I test/externals/ruby1.9/ruby'
     t.ruby_opts << '-r ./test/ruby19_env.rb'
+    t.ruby_opts << '-r minitest'
     t.ruby_opts << '-r minitest/excludes'
   end
   
--- a/bin/rake
+++ b/bin/rake
@@ -19,5 +19,14 @@
   end
 end
 
-gem 'rake', version
-load Gem.bin_path('rake', 'rake', version)
+begin
+  gem 'rake', version
+  load Gem.bin_path('rake', 'rake', version)
+rescue NameError => e # --disable-gems
+  raise unless e.name == :gem
+rescue Gem::LoadError, LoadError
+end
+
+require 'rake'
+
+Rake.application.run
