From: Miguel Landaeta <nomadium@debian.org>
Date: Fri, 29 May 2015 21:11:16 -0300
Subject: Disable tests with unexpected results when jruby is built inside a pbuilder chroot
Forwarded: no

diff --git a/spec/ruby/core/dir/home_spec.rb b/spec/ruby/core/dir/home_spec.rb
index 23307c8..a1da921 100644
--- a/spec/ruby/core/dir/home_spec.rb
+++ b/spec/ruby/core/dir/home_spec.rb
@@ -15,9 +15,9 @@ ruby_version_is "1.9" do
       Dir.home.should == home_directory
     end
 
-    it "returns the named user's home directory as a string if called with an argument" do
-      Dir.home(ENV['USER']).should == home_directory
-    end
+    #it "returns the named user's home directory as a string if called with an argument" do
+    #  Dir.home(ENV['USER']).should == home_directory
+    #end
 
     it "raises an ArgumentError if the named user doesn't exist" do
       lambda { Dir.home('geuw2n288dh2k') }.should raise_error(ArgumentError)
diff --git a/spec/ruby/core/file/expand_path_spec.rb b/spec/ruby/core/file/expand_path_spec.rb
index 4fa6855..4c8b164 100644
--- a/spec/ruby/core/file/expand_path_spec.rb
+++ b/spec/ruby/core/file/expand_path_spec.rb
@@ -130,10 +130,10 @@ describe "File.expand_path" do
       lambda { File.expand_path("~a_not_existing_user") }.should raise_error(ArgumentError)
     end
 
-    it "expands ~ENV['USER'] to the user's home directory" do
-      File.expand_path("~#{ENV['USER']}").should == ENV['HOME']
-      File.expand_path("~#{ENV['USER']}/a").should == "#{ENV['HOME']}/a"
-    end
+    #it "expands ~ENV['USER'] to the user's home directory" do
+    #  File.expand_path("~#{ENV['USER']}").should == ENV['HOME']
+    #  File.expand_path("~#{ENV['USER']}/a").should == "#{ENV['HOME']}/a"
+    #end
 
     it "does not expand ~ENV['USER'] when it's not at the start" do
       File.expand_path("/~#{ENV['USER']}/a").should == "/~#{ENV['USER']}/a"
diff --git a/spec/ruby/library/etc/getlogin_spec.rb b/spec/ruby/library/etc/getlogin_spec.rb
index a217ae1..914e16c 100644
--- a/spec/ruby/library/etc/getlogin_spec.rb
+++ b/spec/ruby/library/etc/getlogin_spec.rb
@@ -2,13 +2,13 @@ require File.expand_path('../../../spec_helper', __FILE__)
 require 'etc'
 
 describe "Etc.getlogin" do
-  it "returns the name of the user who runs this process" do
-    if Etc.getlogin
-      Etc.getlogin.should == username
-    else
-      # Etc.getlogin may return nil if the login name is not set
-      # because of chroot or sudo or something.
-      Etc.getlogin.should be_nil
-    end
-  end
+  #it "returns the name of the user who runs this process" do
+  #  if Etc.getlogin
+  #    Etc.getlogin.should == username
+  #  else
+  #    # Etc.getlogin may return nil if the login name is not set
+  #    # because of chroot or sudo or something.
+  #    Etc.getlogin.should be_nil
+  #  end
+  #end
 end
